test/parallel/test-http-server-keepalive-req-gc.js

```diff
@@ -0,0 +1,42 @@
+// Flags: --expose-gc
+'use strict';
+const common = require('../common');
+const onGC = require('../common/ongc');
+const { createServer } = require('http');
+const { connect } = require('net');
+
+// Make sure that for HTTP keepalive requests, the req object can be
+// garbage collected once the request is finished.
+// Refs: https://github.com/nodejs/node/issues/9668
+
+let client;
+const server = createServer(common.mustCall((req, res) => {
+  onGC(req, { ongc: common.mustCall() });
+  req.resume();
+  req.on('end', common.mustCall(() => {
+    setImmediate(() => {
+      client.end();
```

Comment:
Is this needed?

Reply:
It’s what makes the test finish, because otherwise the connection remains ref’ed and active on the server side, yes. Using `.write()` instead of `.end()` below to write the data to the socket is part of the test here

Reply:
Ok, makes sense. Thanks.